github/resource_github_repository_tag_protection.go

```diff
@@ -0,0 +1,118 @@
+package github
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/google/go-github/v47/github"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func resourceGithubRepositoryTagProtection() *schema.Resource {
```

Comment:
Do you mind justifying why you chose a new resource rather than adding onto an existing one, such as `github_repository`? 

I think it's the right call, but I'm having trouble verbalizing it.

Reply:
Sure, I think that having it as its own resource keeps it in line with how the branch_protection resource is separate. Although the Tags API has fairly basic functionality right now I can see it growing in features over time and I think being able to manage that separately will give us greater flexibility.